core/payment/src/processor.rs

```diff
@@ -292,14 +292,14 @@ impl PaymentProcessor {
             .await??;
 
         // Verify if amount declared in message matches actual amount transferred on blockchain
-        if &details.amount != &payment.amount {
+        if &details.amount < &payment.amount {
```

Comment:
I believe we should rather aim for having the exact amount sent in `SendPayment` message. Instead, we can put `<` here: https://github.com/golemfactory/yagna/blob/60ecce975d10a7fb9ace34540a02f8f54cc04947/core/payment/src/processor.rs#L53